### ISSUE TITLE
`finalMinLevel` not compatible with environment variables

### ISSUE THREAD

**Issue by Aaronmsv** [NLog/NLog#5172]
I want to configure something similar to the default `appsettings.json` in NLog. I'm using `finalMinLevel` do this, but it seems incompatible with environment variables.

**NLog version**: 5.1.0

**Platform**: .NET6

**Current NLog config**

```xml
  <rules>
    <!-- Only show Microsoft logs from 'Warn' level or higher because it spams too much on 'Info',
         except for Microsoft.Hosting.Lifetime which doesn't spam and has useful information.
         This is the ASP.NET default configuration. -->
    <logger name="Microsoft.*" finalMinLevel="${environment:HOST_LOG_LEVEL}" />
    <logger name="Microsoft.Hosting.Lifetime" finalMinLevel="${environment:APP_LOG_LEVEL}" />
    <logger name="*" minlevel="${environment:APP_LOG_LEVEL}" maxLevel="Warn" writeTo="stdout" />
    <logger name="*" minlevel="Error" writeTo="stderr" />
  </rules>
```

**What is the current result?**

Nothing gets logged. When I configure NLog to throw errors, I get the following exception:

> Unhandled exception. System.TypeInitializationException: The type initializer for 'MyCompany.Configuration.ConfigurationBuilderExtensions' threw an exception.
 ---> NLog.NLogConfigurationException: Exception when loading configuration MyProject/bin/Debug/net6/NLog.config
 ---> System.ArgumentException: Unknown log level: ${environment:HOST_LOG_LEVEL} (Parameter 'levelName')
   at NLog.LogLevel.FromString(String levelName)
   at NLog.Config.LoggingConfigurationParser.LogLevelFromString(String text)
   at NLog.Config.LoggingConfigurationParser.ParseRuleElement(ValidatedConfigurationElement loggerElement)
   at NLog.Config.LoggingConfigurationParser.ParseRulesElement(ValidatedConfigurationElement rulesElement, IList`1 rulesCollection)
   at NLog.Config.LoggingConfigurationParser.LoadConfig(ILoggingConfigurationElement nlogConfig, String basePath)
   at NLog.Config.XmlLoggingConfiguration.ParseNLogElement(ILoggingConfigurationElement nlogElement, String filePath, Boolean autoReloadDefault)
   at NLog.Config.XmlLoggingConfiguration.ParseTopLevel(NLogXmlElement content, String filePath, Boolean autoReloadDefault)
   at NLog.Config.XmlLoggingConfiguration.Initialize(XmlReader reader, String fileName, Boolean ignoreErrors)
   --- End of inner exception stack trace ---
   at NLog.Config.XmlLoggingConfiguration.Initialize(XmlReader reader, String fileName, Boolean ignoreErrors)
   at NLog.Config.XmlLoggingConfiguration..ctor(XmlReader reader, String fileName, LogFactory logFactory)
   at NLog.Config.LoggingConfigurationFileLoader.LoadXmlLoggingConfiguration(XmlReader xmlReader, String configFile, LogFactory logFactory)
   at NLog.Config.LoggingConfigurationFileLoader.LoadXmlLoggingConfigurationFile(LogFactory logFactory, String configFile)
   at NLog.Config.LoggingConfigurationFileLoader.TryLoadLoggingConfiguration(LogFactory logFactory, String configFile, LoggingConfiguration& config)
   at NLog.Config.LoggingConfigurationFileLoader.TryLoadFromFilePaths(LogFactory logFactory, String filename)
   at NLog.Config.LoggingConfigurationFileLoader.Load(LogFactory logFactory, String filename)
   at NLog.Config.LoggingConfigurationWatchableFileLoader.Load(LogFactory logFactory, String filename)
   at NLog.LogFactory.get_Configuration()
   at NLog.LogFactory.GetLoggerThreadSafe(String name, Type loggerType)
   at NLog.LogFactory.GetLogger(String name)
   at NLog.LogManager.GetLogger(String name)

**What is the expected result?**

Log levels are retrieved from the environmental variables and NLog logs as the logs as expected.

**Are there any workarounds?**

It is not exactly the same, but close enough, and this seems to work just fine:

```xml
  <rules>
    <!-- Only show Microsoft logs from 'Warn' level or higher because it spams too much on 'Info',
         except for Microsoft.Hosting.Lifetime which doesn't spam and has useful information.
         This is the ASP.NET default configuration. -->
    <logger name="Microsoft.Hosting.Lifetime" minLevel="${environment:APP_LOG_LEVEL}" maxLevel="Warn" writeTo="stdout" final="true" />
    <logger name="Microsoft.Hosting.Lifetime" minLevel="Error" writeTo="stderr" final="true" />
    <logger name="Microsoft.Hosting.Lifetime" maxLevel="${environment:APP_LOG_LEVEL}" final="true" />
    <logger name="Microsoft.*" minLevel="${environment:HOST_LOG_LEVEL}" maxLevel="Warn" writeTo="stdout" final="true" />
    <logger name="Microsoft.*" minLevel="Error" writeTo="stderr" final="true" />
    <logger name="Microsoft.*" maxLevel="${environment:HOST_LOG_LEVEL}" final="true" />
    <logger name="*" minlevel="${environment:APP_LOG_LEVEL}" maxLevel="Warn" writeTo="stdout" />
    <logger name="*" minlevel="Error" writeTo="stderr" />
  </rules>
```

**Is there a version in which it did work?**

I have not tested multiple versions.

**Comment by snakefoot** [comment]
Yes you are right there is currently no Layout-support for `finalMinLevel`.

Pull-requests are ofcourse wellcome.

**Comment by Aaronmsv** [comment]
I'll take a look.

**Comment by snakefoot** [comment]
Resolved by #5174

**Comment by snakefoot** [comment]
NLog v5.1.3 has been released - https://www.nuget.org/packages/NLog#release-body-tab
